index.html

```diff
@@ -2992,174 +2882,221 @@ <h3>Other Discovery Methods</h3>
-					<details>
-						<summary>Explanation</summary>
-						<p>The global language declaration obtained here is used to set the base direction for
-							localizable strings without a declaration.</p>
-					</details>
-				</li>
+					<li id="processing-conformance">
+						<p>(<a href="#profile-conformance"></a>) Determine the <a>profile(s)</a> the manifest conforms
+							to as follows:</p>
+						<ol>
+							<li>If <var>manifest["conformsTo"]</var> is not set, this is a <a>fatal error</a>. Return
+								failure.</li>
+							<li>If <var>manifest["conformsTo"]</var> does not include a URL of a profile it is capable
+								of processing and/or rendering, this is a <a>fatal error</a>. Return failure.</li>
+							<li>If <var>manifest["conformsTo"]</var> includes the URLs of more than one profile it is
+								capable of processing and rendering, the user agent MUST select the first in the array
+								in the case of conflicting requirements.</li>
+							<li>Otherwise, the publication is an instance of the profile defined by the recognized
+								URL.</li>
+						</ol>
+						<details>
+							<summary>Explanation</summary>
+							<p>The profile(s) the publication conforms to determines any additional extension steps that
+								have to be performed during processing.</p>
+						</details>
+					</li>
+
+					<li id="processing-lang-dir">
+						<p>(<a href="#manifest-lang-dir-global"></a>) Let <var>lang</var> be the global language and
+								<var>dir</var> be the global direction obtained from this step. Set each initially to an
+							empty <a href="https://infra.spec.whatwg.org/#string">string</a>.</p>
+						<p><a href="https://infra.spec.whatwg.org/#map-iterate">For each</a>
+							<var>context</var> in <var>manifest["@context"]</var>, moving from the last <a
+								href="https://infra.spec.whatwg.org/#struct-item">item</a> to the first:</p>
+						<ol>
+							<li>if <var>lang</var> is an empty <a href="https://infra.spec.whatwg.org/#string"
+									>string</a> and <var>context</var> is a <a
+									href="https://infra.spec.whatwg.org/#ordered-map">map</a> that contains a
+									<var>language</var>
+								<a href="https://infra.spec.whatwg.org/#map-key">key</a>, set <var>lang</var> to
+									<var>context["language"]</var>;</li>
+							<li>if <var>dir</var> is an empty <a href="https://infra.spec.whatwg.org/#string">string</a>
+								and <var>context</var> is a <a href="https://infra.spec.whatwg.org/#ordered-map">map</a>
+								that contains a <var>direction</var>
+								<a href="https://infra.spec.whatwg.org/#map-key">key</a>, set <var>dir</var> to
+									<var>context["direction"]</var>;</li>
+							<li>if both <var>lang</var> and <var>dir</var> are not empty <a
+									href="https://infra.spec.whatwg.org/#string">strings</a>, then <a
+									href="https://infra.spec.whatwg.org/#iteration-break">break</a>.</li>
+						</ol>
```

Comment:
I am not sure that is correct. If the contexts are:

```json
[{ "language": "en", "direction":"ltr"}, {"language":"ar"}]
```

then the global language will be set to `en`, because it is only getting at that point that _both_ language and dir are not empty, i.e., a break would happen.

I have the impression that it is cleaner to (conceptually) run the cycles for language and for direction independently of one another, and not trying to merge them.

Reply:
No, this wouldn't happen because neither `lang` nor `dir` gets set again once they get a value.

Working backwards, `lang` gets set to "ar", so is no longer an empty string, and `dir` is and empty string after inspecting the first object. Then `dir` gets set to "ltr" from the next object and neither is an empty string any longer so the process ends.

Reply:
Yep, you are right!